### PR TITLE
CI: add docformatter to precommit

### PR DESCRIPTION
**Pull request by Borda**
let's get unify the docstrings

cc: @EnricoMi

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,34 +5,44 @@ repos:
...
       - id: pyupgrade
-        args:
-          - --py37-plus
+        args: ["--py37-plus"]
+
+  - repo: https://github.com/PyCQA/docformatter
+    rev: v1.7.3
+    hooks:
+      - id: docformatter
+        args: ["--in-place"]
```
Comment (trim21) [inline .pre-commit-config.yaml:44]: Also use `--black` flag

**Review by trim21 — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Borda) [reply .pre-commit-config.yaml:44]:
```suggestion
        args: ["--in-place", "--black"]
```

**Comment by Borda** [comment]
@sfdye @EnricoMi could you pls have look / review :flamingo:

File: github/RepositoryAdvisoryVulnerabilityPackage.py [hunk 2]
```diff
@@ -26,11 +26,10 @@
 from github.GithubObject import Attribute, NonCompletableGithubObject, NotSet
 
 
-class RepositoryAdvisoryVulnerabilityPackage(github.GithubObject.NonCompletableGithubObject):
-    """This class represents an identifier for a package that is vulnerable to a parent SecurityAdvisory.
+class RepositoryAdvisoryVulnerabilityPackage(NonCompletableGithubObject):
+    """This class represents an identifier for a package that is vulnerable tao  parent SecurityAdvisory.
```
Comment (EnricoMi) [inline github/RepositoryAdvisoryVulnerabilityPackage.py:30]: What is this? An attempt to correct spelling?

**Review by EnricoMi — CHANGES_REQUESTED** [review]
I general, the formatting changes look good to me, but there is some unexpected spelling change.

File: github/Repository.py [hunk 3]
```diff
@@ -2349,18 +2373,27 @@ def delete_file(
...
+            repository’s default branch (usually master)
+        :param committer: InputGitAuthor, (optional), if no information
+            is given the authenticated user's information will be used.
+            You must specify both a name and email.
+        :param author: InputGitAuthor, (optional), if omitted this will
+            be filled in with committer information. If passed, you must
+            specify both a name and email.
+        :rtype: { 'content'
+        : : class:`null <github.GithubObject.NotSet>`:, 'commit'
+        : : class:`Commit <github.Commit.Commit>`}
```
Comment (Borda) [inline github/Repository.py:2395]: ok, this also seems wrong

File: github/RepositoryAdvisoryVulnerabilityPackage.py (hunk 2 above)
Comment (Borda) [reply github/RepositoryAdvisoryVulnerabilityPackage.py:30]: well it is the same as the original, right, so no docformater issue...

File: github/Repository.py (hunk 3 above)
Comment (Borda) [reply github/Repository.py:2395]: ok, I think in this case the docstring shall be marked with r to be parsed as RST

**Comment by trim21** [comment]
image

still doesn't look good

**Comment by trim21** [comment]
image

looks acceptable to me

**Comment by trim21** [comment]
> > image
> > if the line break in url is not a problem, looks acceptable to me
> 
> Seems like your message is unrelated to actual state, please update it accordingly

I'm running pre-commit on all files at latest commit b30ea575 of this PR.

**Comment by Borda** [comment]
> I'm running pre-commit on all files at latest commit [b30ea57](https://github.com/PyGithub/PyGithub/commit/b30ea575432c31b80c0a101926ac99a7c56486e7) of this PR.

Cool and have you noticed the PR status?

**Comment by Borda** [comment]
@EnricoMi applied after all the pyi were merged and all seems fine now, pls check it again :)

**Comment by Borda** [comment]
@EnricoMi mind pls checking again :pray:

**Comment by Borda** [comment]
@EnricoMi is there anything I can do to help it land? 🦩

File: github/MainClass.py [hunk 4]
```diff
@@ -757,38 +755,59 @@ def get_emojis(self) -> dict[str, str]:
...
+        :param obj: the object to pickle
+        :param file: the file-like object to pickle to
+        :param protocol: the `pickling protocol <https://python.readthedocs.io/en/latest/library/pickle.html#data-
+            stream-format>`_
         :param obj: the object to pickle
         :param file: the file-like object to pickle to
-        :param protocol: the `pickling protocol <https://python.readthedocs.io/en/latest/library/pickle.html#data-stream-format>`_
+        :param protocol: the `pickling protocol <https://python.readthedocs.io/en/latest/library/pickle.html#data-
+            stream-format>`_
+
```
Comment (EnricoMi) [inline github/MainClass.py:801]: This does not look right.

File: github/ApplicationOAuth.py [hunk 5]
```diff
@@ -37,9 +37,11 @@
 
 
 class ApplicationOAuth(NonCompletableGithubObject):
-    """
-    This class is used for identifying and authorizing users for Github Apps.
-    The reference can be found at https://docs.github.com/en/developers/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps
+    """This class is used for identifying and authorizing users for Github Apps.
```
Comment (EnricoMi) [inline github/ApplicationOAuth.py:40]: Can we keep the first line below `"""`? Is this configurable?

**Review by EnricoMi — CHANGES_REQUESTED** [review]

File: github/ApplicationOAuth.py (hunk 5 above)
Comment (Borda) [reply github/ApplicationOAuth.py:40]: I do not think so, as the standard is to have online and then the remaining docs...
Any reason for this special formatting, just personal preference?

Comment (Borda) [reply github/ApplicationOAuth.py:45]: @EnricoMi ?

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -34,6 +34,13 @@ repos:
       - id: codespell
         additional_dependencies: [tomli]
 
+  - repo: https://github.com/PyCQA/docformatter
+    rev: v1.7.5
+    hooks:
+      - id: docformatter
+        additional_dependencies: [tomli]
+        args: ["--in-place"]
```
Comment (EnricoMi) [inline .pre-commit-config.yaml:42]: Looks like there are the necessary options:
```suggestion
        args: ["--in-place", "--pre-summary-newline", "--make-summary-multi-line"]
```
This is to keep the changes of this PR to the docs at a minimum.

**Comment by Borda** [comment]
@EnricoMi mind kindly review again, pls :pray:

**Review by EnricoMi — APPROVED** [review]

**Comment by EnricoMi** [comment]
Please rebase and rerun again.

**Comment by Borda** [comment]
> Please rebase and rerun again.

@EnricoMi merged the `main` and rerun pre-commit with no changes
our you really need to rebase it? :flamingo:

**Review by EnricoMi — APPROVED** [review]

**Comment by Borda** [comment]
@EnricoMi thank you, it was long and really appreciate your patience =)

**Pull request merged**